Rakefile

```diff
@@ -151,6 +151,24 @@ namespace :build do
         HipChat.log 'Migrating <b>dashboard</b> database...'
         RakeUtils.rake 'db:migrate'
 
+        # Update the schema cache file, except for production which always uses the cache.
+        schema_cache_file = dashboard_dir('db/schema_cache.dump')
+        unless rack_env?(:production)
+          RakeUtils.rake 'db:schema:cache:dump'
+          if RakeUtils.file_changed_from_git?(schema_cache_file)
+            # Staging is responsible for committing the authoritative schema cache dump.
+            if rack_env?(:staging)
+              RakeUtils.system 'git', 'add', schema_cache_file
```

Comment:
I think the add is still needed.  An experiment suggests that you still need to add the file before you can commit it, even when you specify it on the command line for the commit:

$ echo "hello" > foo.txt
$ git commit -m "add foo" foo.txt
error: pathspec 'foo.txt' did not match any file(s) known to git.